org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/TableAttributePanel.java

```diff
@@ -0,0 +1,373 @@
+package org.adempiere.webui.panel;
+
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+
+import org.adempiere.webui.apps.AEnv;
+import org.adempiere.webui.component.ConfirmPanel;
+import org.adempiere.webui.component.Grid;
+import org.adempiere.webui.component.Label;
+import org.adempiere.webui.component.Row;
+import org.adempiere.webui.component.Rows;
+import org.adempiere.webui.component.Urlbox;
+import org.adempiere.webui.component.Window;
+import org.adempiere.webui.editor.WEditor;
+import org.adempiere.webui.editor.WebEditorFactory;
+import org.adempiere.webui.event.ValueChangeEvent;
+import org.adempiere.webui.event.ValueChangeListener;
+import org.adempiere.webui.session.SessionManager;
+import org.adempiere.webui.util.ZKUpdateUtil;
+import org.compiere.model.GridField;
+import org.compiere.model.GridFieldVO;
+import org.compiere.model.MAttribute;
+import org.compiere.model.MAttributeSet;
+import org.compiere.model.MTable;
+import org.compiere.model.MTableAttribute;
+import org.compiere.model.PO;
+import org.compiere.util.CLogger;
+import org.compiere.util.DisplayType;
+import org.compiere.util.Env;
+import org.compiere.util.Msg;
+import org.compiere.util.Util;
+import org.zkoss.zk.ui.Component;
+import org.zkoss.zk.ui.event.Event;
+import org.zkoss.zk.ui.event.EventListener;
+import org.zkoss.zul.Vbox;
+
+public class TableAttributePanel extends Window implements EventListener<Event>
+{
+	/**
+	 * generated serial id
+	 */
+	private static final long	serialVersionUID	= -4922961793415942591L;
+
+	/** the attribute set selected on the InfoProduct window */
+	private int					p_M_AttributeSet_ID	= 0;
+	private int					p_AD_Table_ID		= 0;
+	private int					p_Record_ID			= 0;
+	protected int				m_WindowNo;
+
+	/**
+	 * Constructor.
+	 * Called from AbstractADWindowContent.onAttributeForm(),
+	 * 
+	 * @param tableID
+	 * @param recordID
+	 */
+	public TableAttributePanel(Integer tableID, int recordID)
+	{
+		super();
+		p_AD_Table_ID = tableID;
+		p_Record_ID = recordID;
+		p_M_AttributeSet_ID = MTable.get(Env.getCtx(), tableID).getM_AttributeSet_ID();
+		setTitle(Msg.getMsg(Env.getCtx(), "InfoPAttribute"));
+		this.setBorder("normal");
+		this.setMaximizable(false);
+		this.setSizable(false);
+		m_WindowNo = SessionManager.getAppDesktop().registerWindow(this);
+
+		try
+		{
+			jbInit();
+			dynInit();
+		}
+		catch (Exception e)
+		{
+			log.log(Level.SEVERE, "InfoPAttribute", e);
+		}
+		AEnv.showWindow(this);
+	} // InfoPAttribute
+
+	/** Attribute Editors */
+	private Map<Integer, WEditor>	m_attEditors	= new HashMap<Integer, WEditor>();
+	/** Logger */
+	private static final CLogger	log				= CLogger.getCLogger(TableAttributePanel.class);
+
+	private Rows					rows			= null;
+	private ConfirmPanel			confirmPanel	= new ConfirmPanel(true);
+
+	/**
+	 * Layout dialog
+	 * 
+	 * @throws Exception
+	 */
+	private void jbInit() throws Exception
```

Comment:
understood that this is from copy and paste but it is still weird to have jbInit naming in new code.